### PR TITLE
major updates - code reorg, c4.5 efficiency improvements, minor vis update

### PR DESCRIPTION
**Pull request by yangdanny97**
Vis:
- looks somewhat cleaner, displays the accuracy of validation set automatically when a tree is selected; highlighted features are preserved when tree is switched. 
- next steps are to make aggregate view of trees sorted by accuracy, with ability to superimpose feature importance info

Code reorg:
- all cleaning/pos checking functions are moved to cleaning.py
- idf/body processing remains in preprocessing.py (because these are actually preprocessed before feature generation)
- sentence features and final feature generation moved to feature_engineering.py
- word embedding functions moved to word_embeddings.py
- NOTE: preprocessing remains the interface to this little library, if you need to access something just add a stub in preprocessing (examples at bottom of file)

c4.5 rework: (partial list, IDR all the changes)
- removed lambdas to reduce function call overhead (minor)
- used counters instead of custom dictionary function (minor)
- deduplicated column values when choosing splitting points (major)
- added max depth, min leaf size, min splitting size params 

training time on 4000 data points is in single digit seconds; likely training on our 40k training set would take 2 mins/tree. added some helper functions to test runtime/accuracy in the ipynb for convenience, otherwise all implementation moved to c45.py

future work would be writing code to make JSON dump of the c4.5 tree

### CODE REVIEW

**Comment by maxlchen** [comment]
Changes look clean
